parse.js

```diff
@@ -6,40 +6,33 @@
 	License: Apache-2.0
 */
 
-const ARGUMENT_SEPARATION_REGEX = /([^=\s]+)=?\s*(.*)/;
+const ARGUMENT_SEPARATION_REGEX = /^([^=\s]+)=?\s*(.*)/;
 
 function Parse(argv) {
-  // Removing node/bin and called script name
-  argv = argv.slice(2);
-
+  
   const parsedArgs = {};
-  let argName, argValue;
-
-  argv.forEach(function (arg) {
-    // Separate argument for a key/value return
-    arg = arg.match(ARGUMENT_SEPARATION_REGEX);
-    arg.splice(0, 1);
-
-    // Retrieve the argument name
-    argName = arg[0];
-
+  // Removing node/bin and called script name by starting at index 2
+  for(let i = 2; i < argv.length; i++){
+    // Separate argument for a key/value return, ignoring full match
+    let [_, argName, argValue] = argv[i].match(ARGUMENT_SEPARATION_REGEX);
+    
     // Remove "--" or "-"
-    if (argName.indexOf('-') === 0) {
-      argName = argName.slice(argName.slice(0, 2).lastIndexOf('-') + 1);
+    if (argName[0] === '-') {
+      argName = argName.slice(argName[1] === '-' ? 2 : 1);
     }
-
+    
     // Parse argument value or set it to `true` if empty
-    argValue =
-      arg[1] !== ''
-        ? parseFloat(arg[1]).toString() === arg[1]
-          ? +arg[1]
-          : arg[1]
+    const numberValue = +argValue
+    parsedArgs[argName] =
+      argValue !== ''
+        ? !Number.isNaN(numberValue) ? numberValue : argValue
         : true;
```

Comment:
- It's fine to repeat `+argValue`, no need to allocate a variable :)
- Positive conditions are usually easier to read:
```js
if (true) {
// vs
if (!true) {
```
You could rewrite this block with:

```js
parsedArgs[argName] =
  argValue !== ''
    ? Number.isNaN(+argValue) ? argValue : +argValue
    : true;
```

Reply:
Thanks for pointing that out. I actually wrote the fully optimized and minified code first, and then attempted expanding it, explaining the reversed order. Whoops 😅 

One thing that I must point out is your wording of "allocating" a variable. Modern CPUs have enough registers that likely no variable is actually being "allocated". In fact, by the repeated integer parsing operation, the repeated option is negligibly _slower_. Not important in a package like this, but just keep in mind.